operations/validate.go

```diff
@@ -129,26 +135,40 @@ func validateFile(path string, ac *legacyClient, quiet, includeLong bool, localM
 		return nil
 	}
 
-	if len(projErrors) != 0 {
-		grip.Info(projErrors)
-	}
-	hasErrs := false
-	hasWarnings := false
-	for _, projErr := range projErrors {
-		if projErr.Level == validator.Error {
-			hasErrs = true
-		} else if projErr.Level == validator.Warning {
-			hasWarnings = true
-		}
-	}
-
-	if hasErrs {
+	grip.Info(projErrors)
+	if projErrors.HasError() {
 		return errors.Errorf("%s is an invalid configuration", path)
-	} else if hasWarnings {
+	} else if len(projErrors) > 0 {
 		grip.Infof("%s is valid with warnings", path)
 	} else {
 		grip.Infof("%s is valid", path)
 	}
 
 	return nil
 }
+
+// loadProjectIntoWithValidation returns a warning (instead of an error) if there's an error with unmarshalling strictly
+func loadProjectIntoWithValidation(ctx context.Context, data []byte, opts *model.GetProjectOpts,
+	project *model.Project) (*model.ParserProject, validator.ValidationErrors) {
+	errs := validator.ValidationErrors{}
+	pp, _, err := model.LoadProjectInto(ctx, data, opts, "", project)
+	if err != nil {
+		if opts.UnmarshalStrict {
+			// try it without unmarshal strict; if the error is from being strict, warn but don't error.
```

Comment:
Perhaps we can wrap the error returned if UnmarshalYAMLStrictWithFallback fails in createIntermediateProject with a string and index this error for that specific string? Instead of running LoadProjectInto twice and specifying this new err2

Reply:
Good idea -- I don't think we can do away with running LoadProjectInto twice entirely, because UnmarshalStrict can error for both strict and non-strict reasons; we don't know which it is without running the same thing _without_ unmarshal strict (i.e. determining if it should be a warning or an error). You're right though that we can limit this edge case to just the unmarshalling yaml part.